src/elements/hx-pill/index.spec.js

```diff
@@ -0,0 +1,107 @@
+        });
+
+    });
+
+    describe('test Shadow DOM', () => {
+        describe('verify render', () => {
+            it('should have a static Shadow DOM', async function () {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
+                const shadow = component.shadowRoot.innerHTML;
+
+                expect(component).shadowDom.to.equal(shadow);
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
+                const mode = component.shadowRoot.mode;
+
+                expect(mode).to.equal("open");
+            });
+
+            it('should have a single <slot>', async () => {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelectorAll('slot');
+                const len = query.length;
+
+                expect(len).to.be.equal(1);
+            });
+
+            it('should have an unnamed <slot>', async () => {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
+                const name = component.slot;
+
+                if ( name !== null ) {
+                    expect(name).to.be.equal('');
+                } else {
+                    expect(name).to.be.null;
+                }
+            });
+        });
+
+        describe('verify Shadow DOM markup', () => {
+
+            it('markup should contain a #hxDismiss button', async () => {
+                const elSelector = 'button#hxDismiss';
+                const id = 'hxDismiss';
+                const component = /** @type { HXPillEvent } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxDismiss button with a times type <hx-icon>', async () => {
+                const elSelector = 'button#hxDismiss > span > hx-icon';
+                const iconType = 'times';
+                const component = /** @type { HXPillEvent } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
+
+                expect(queryId).to.equal(iconType);
+            });
+        });
+    });
+
+    describe('test getters and setters', () => {
+        it('should be able to persist', async () => {
+            const component = /** @type {HXPillEvent} */ await fixture(template);
+            component.persist = true;
+            const attr = component.hasAttribute('persist');
+
+            expect(attr).to.be.true;
+        });     
+    });
```

Comment:
Please add a test for the `click` event.

https://github.com/HelixDesignSystem/helix-ui/blob/master/src/elements/hx-pill/index.js#L35

Reply:
Changes have been done, please review.